content/blog/2017/2017-02-07-declarative-maven-project.adoc

```diff
@@ -0,0 +1,295 @@
+- tutorial
+- pipeline
+- declarative
+- maven
+- java
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:https://cloudbees.com[CloudBees].
+
+**Declare Your Pipelines!**
+link:/blog/2017/02/03/declarative-pipeline-ga/[Declarative Pipeline 1.0 is here]!
+This is first in a series of blog posts that will show some of cool features of
+link:/doc/book/pipeline/syntax/#declarative-pipeline[Declarative Pipeline].
+For several of these posts, I'll be revisiting some of my
+link:/blog/2016/07/18/pipeline-notifications/[previous posts]
+on using with various plugins with (Scripted) Pipeline,
+and seeing how those are implemented in Declarative Pipeline.
+
+To start though, let's get familiar with the basic structure of a Declarative Pipeline
+by creating a simple pipeline for a Maven-based Java project - the
+link:https://plugins.jenkins.io/junit[Jenkins JUnit plugin].
+We'll create a minimal Declarative Pipeline,
+add the settings needed to run Maven and the JDK,
+and finally we'll actually run Maven to build the plugin.
+
+== Setup
+
+With Declarative, it is still possible to run Pipelines edited directly in the
+Jenkins web UI, but one of the key features of "Pipeline as Code" is
+checking-in and being able to track changes.  For this post, I'm going to use
+the
+link:https://github.com/bitwiseman/junit-plugin/tree/blog/add-declarative-pipeline[`blog/add-declarative-pipeline`]
+branch of
+link:https://github.com/bitwiseman/junit-plugin[my fork] of the JUnit plugin.
+I'm going to setup a Multi-branch Pipeline and point it at my repository.
+
+image::/images/post-images/2017-02-07/junit-config.png[JUnit Multi-branch Pipeline Configuration, role="center"]
+
+I've also set this Pipeline's Git configuration to automatically "clean after
+checkout" and to only keep the ten most recent runs.
+
+== Writing a Minimal Pipeline
+
+As has been said before, Declarative Pipeline provides a more structured,
+"opinionated" way to create Pipelines. I'm going to start by creating a minimal
+Declarative Pipeline and adding it to my branch.  Below is a minimal Pipeline
+(with annotations) that just prints a message:
+
+[pipeline]
+----
+// Declarative //
+pipeline { // <1>
+    agent any // <2> <3>
+    stages { // <4>
+        stage('Build') { // <5>
+            steps { // <6>
+               echo 'This is a minimal pipeline.' // <7>
+            }
+        }
+    }
+}
+// Scripted //
+node { // <2>
+    checkout scm // <3>
+    stage ('Build') { // <4>
+        echo 'This is a minimal pipeline.' // <6>
+    }
+}
+----
+<1> All Declarative Pipelines start with a `pipeline` section.
+<2> Select where to run this Pipeline, in this case "any" agent, regardless of label.
+<3> Declarative automatically performs a checkout of source code on the agent,
+    whereas Scripted Pipeline users must explicitly call `checkout scm`,
+<4> A minimal Declarative Pipeline is defined a series of stages.
```

Comment:
"is defined a series of" -> "is defined as a series of"

Maybe remove "minimal"